state/txindex/indexer.go

```diff
@@ -4,20 +4,24 @@ import (
 	"errors"
 
 	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tmlibs/pubsub/query"
 )
 
 // TxIndexer interface defines methods to index and search transactions.
 type TxIndexer interface {
 
-	// AddBatch analyzes, indexes or stores a batch of transactions.
-	// NOTE: We do not specify Index method for analyzing a single transaction
-	// here because it bears heavy performance losses. Almost all advanced indexers
-	// support batching.
-	AddBatch(b *Batch) error
+	// AddBatch analyzes, indexes and stores a batch of transactions.
+	AddBatch(b *Batch, allowedTags []string) error
```

Comment:
would it be better if the `allowedTags` were registered once on the Indexer, rather than with each call ?